src/ast/effects.h

```diff
@@ -164,10 +169,24 @@ struct EffectAnalyzer : public PostWalker<EffectAnalyzer> {
   }
   void visitLoad(Load *curr) {
     readsMemory = true;
+    isAtomic = curr->isAtomic;
```

Comment:
The flag would only change if the instruction itself changed after a previous visit. Shouldn't we keep it up to date with the instruction?